components/sup/src/service_config.rs

```diff
@@ -196,6 +197,11 @@ impl ServiceConfig {
                 debug!("Configuration {} does not exist; restarting", filename);
                 outputln!("Updated {}", Purple.bold().paint(config));
                 self.config_hash.insert(filename.clone(), file_hash);
+                let parent_path = Path::new(&filename).parent();
+                match parent_path {
```

Comment:
for a match where you only care about one pattern (the `Some` in this case), you can use an `if let`:

```
if let Some(pp) = Path::new(&filename).parent {
  try!(create_dir_all(pp));
}
```